### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by Kalunge**
## React capstone project - Metrics web app
in this project I:
- built a mobile web application to check a list of metrics (numeric values) that you will create making use of React and Redux.
- built the homepage with a list of books sorted by the author and showing the number of chapters In each
- Built a details page showing verses in the first chapter of each book
- Created unit tests for pure functions in your Redux code.
- Created integration tests for your application using the React Testing Library.
- created a redux store to hold The data retrieved from the API
- deployed the app on [netlify](https://elated-hawking-3a7b66.netlify.app/)
- recorded a loom [video](https://www.loom.com/share/2e0b981bc05f459f89161283c741b1e1) to showcase the app

### CODE REVIEW

File: src/components/Books.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import { FaBible } from 'react-icons/fa';
+import { authenticate, getBooks } from '../redux/books/books';
+// import bible from '../img/bible.jpg';
```
Comment (Dipeshtwis) [inline src/components/Books.js]: - [ ] I would recommend removing the commented code to make the codebase cleaner.

File: src/__test__/Book.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+it('renders correctly', () => {
+  const tree = renderer
+    .create(
+      <Provider store={store}>
+        <Book book={book} />
+      </Provider>,
+    )
+    .toJSON();
+  expect(tree).toMatchSnapshot();
```
Comment (Dipeshtwis) [inline src/__test__/Book.test.js:16]: - [ ] Some of the tests are failing. Please fix that. :slightly_smiling_face: 
Screenshot from 2021-10-14 14-24-37

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Kalunge, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: App is deployed in netlify.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️
 
 **_Check the comments under the review_**

- [ ] You have done a great job by giving a presentation, but make sure video contains a highlight about a piece of the code the student finds interesting or is proud of making.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/Book.test.js (hunk 2 above)
Comment (Kalunge) [reply src/__test__/Book.test.js:16]: Thanks for the review, I have updated the tests, removed the commented code and also updated the video to include a feature I enjoyed doing.

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @Kalunge, 

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**